### PR TITLE
Feature/ab#26065 import gos data

### PR DESCRIPTION
**Pull request by jimmyPasta**
New API to retrieve Unity Applicants:
Get request - has tenant ID
Lookup by name
https://localhost:44342/api/chefs/applicant/3a14213d-8d0a-21dc-56d6-b62d3a2413ee?UnityApplicantName=Larch Hills Nordic Society
Or --- Lookup by Id
https://localhost:44342/api/chefs/applicant/3a14213d-8d0a-21dc-56d6-b62d3a2413ee?UnityApplicantId=103974

Returns:
{
    "Id": "b8697212-792c-4da8-ad1e-0472df0fd35c",
    "ApplicantName": "Larch Hills Nordic Society",
    "UnityApplicantId": "103974",
    "BcSocietyNumber": "S0015792",
    "OrgNumber": "S0015792",
    "Sector": "ART_CU_SPO",
    "OperatingStartDate": "1980-08-07",
    "FiscalYearDay": "30",
    "FiscalYearMonth": "Sep",
    "BusinessNumber": "854814605",
    "PyhsicalAddressUnit": "",
    "PyhsicalAddressLine1": "378 Thompson Road",
    "PyhsicalAddressLine2": "",
    "PyhsicalAddressPostal": "V2J 5B5",
    "PyhsicalAddressCity": "Quesnel",
    "PyhsicalAddressProvince": "BC",
    "PyhsicalAddressCountry": "Canada",
    "PhoneNumber": "(250) 992-9211",
    "PhoneExtension": ""
}

Also added Orgbook service and matching:
                            string nameText = name.SelectToken("text").ToString();
                            double match = await nameText.CompareStringsAsync(applicant.ApplicantName);
                            
                            Added in string comparison for importing Orgs and matching against orgbook

### CODE REVIEW

File: applications/Unity.GrantManager/modules/Unity.SharedKernel/Utils/LetterPairGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            }
+        }
+
+        // Async method to split the string with a timeout using a CancellationToken
+        private static string[] SplitWithTimeout(string str, int timeoutSeconds)
+        {
+
+            // Start a task to perform the Regex.Split asynchronously
+#pragma warning disable SYSLIB1045
+            var task = Regex.Split(str, @"\s+", RegexOptions.None, TimeSpan.FromSeconds(timeoutSeconds));
```
Comment (AndreGAot) [inline applications/Unity.GrantManager/modules/Unity.SharedKernel/Utils/LetterPairGenerator.cs:31]: I think you may have changed something here at some point - Visual Studio believes that the pragma is not required. - also there is nothing from the outside of this method call to indicate that a task is being returned - no async , cancellation etc..

File: applications/Unity.GrantManager/src/Unity.GrantManager.Application/Integrations/OrgBook/OrgBookService.cs [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+using Volo.Abp;
+
+namespace Unity.GrantManager.Integrations.Orgbook
+{
+    [IntegrationService]
+    public class OrgBookService : GrantManagerAppService, IOrgBookService
+    {
+        private readonly IResilientHttpRequest _resilientRestClient;
+
+        private readonly string orgbook_base_api = "https://orgbook.gov.bc.ca/api/v4";
```
Comment (AndreGAot) [inline applications/Unity.GrantManager/src/Unity.GrantManager.Application/Integrations/OrgBook/OrgBookService.cs:16]: this should be moved to appsettings.json

File: applications/Unity.GrantManager/src/Unity.GrantManager.Domain/Intakes/ApplicantResult.cs [hunk 3]
```diff
@@ -4,31 +4,24 @@ namespace Unity.GrantManager.Intakes
 {
     public class ApplicantResult
     {
-        [JsonPropertyName("id")]
```
Comment (AndreGAot) [inline applications/Unity.GrantManager/src/Unity.GrantManager.Domain/Intakes/ApplicantResult.cs:7]: I feel like there was a specific reason that we had explicit json property names for these fields - just double checking that its not relevant anymore

File: applications/Unity.GrantManager/src/Unity.GrantManager.Application/Integrations/OrgBook/OrgBookService.cs (hunk 2 above)
Comment (jimmyPasta) [reply applications/Unity.GrantManager/src/Unity.GrantManager.Application/Integrations/OrgBook/OrgBookService.cs:16]: Daryl does not want them in App settings, I guess I can add this to the configuration but it is not going to change.

File: applications/Unity.GrantManager/modules/Unity.SharedKernel/Utils/LetterPairGenerator.cs (hunk 1 above)
Comment (jimmyPasta) [reply applications/Unity.GrantManager/modules/Unity.SharedKernel/Utils/LetterPairGenerator.cs:31]: I think sonar qube had issues with the @"\s+" not being in a [RegularExpression(@"\s+""
Annotation. I added the Annotation but then in the SharedKernal it did not have the Nuget package.
I installed the Nuget package for <PackageReference Include="System.Text.RegularExpressions" Version="4.3.1" />
into the Shared Kernel proj but it then complained:
Error (active)	CS0246	The type or namespace name 'GeneratedRegexAttribute' could not be found (

I will take out the pragma now and see if it complains.
One of the issues was around not having a timeout which I fixed, that was where all the awaits came in as I originally had pushed it into a Run with a cancel token timeout, the Timespan Regex timeout is cleaner though

File: applications/Unity.GrantManager/src/Unity.GrantManager.Domain/Intakes/ApplicantResult.cs (hunk 3 above)
Comment (jimmyPasta) [reply applications/Unity.GrantManager/src/Unity.GrantManager.Domain/Intakes/ApplicantResult.cs:7]: It is returning fine without it

File: applications/Unity.GrantManager/src/Unity.GrantManager.Application/Integrations/OrgBook/OrgBookService.cs (hunk 2 above)
Comment (AndreGAot) [reply applications/Unity.GrantManager/src/Unity.GrantManager.Application/Integrations/OrgBook/OrgBookService.cs:16]: we can leave it for now, but definitely tech debt and should be looked at with the next tech debt sprint

**Review by AndreGAot — APPROVED** [review]
Ready for Dev - approved on the premise that we look at the orgbook connection string during next tech debt sprint, and move the connection strings to the db where appropriate

**Pull request merged**